rust/relay/ebpf-turn-router/src/main.rs

```diff
@@ -1,434 +1,475 @@
-    remove_channel_data_header_ipv6,
-};
-use network_types::{
-    eth::EtherType,
-    ip::{IpProto, Ipv4Hdr, Ipv6Hdr},
-    udp::UdpHdr,
-};
-use udp::Udp;
+#![cfg_attr(all(target_os = "linux", target_arch = "bpf"), no_std)]
+#![cfg_attr(all(target_os = "linux", target_arch = "bpf"), no_main)]
 
-mod channel_data;
-mod checksum;
-mod config;
-mod error;
-mod eth;
-mod ip4;
-mod ip6;
-mod move_payload;
-mod ref_mut_at;
-mod stats;
-mod udp;
-
-const NUM_ENTRIES: u32 = 0x10000;
-
-// SAFETY: Testing has shown that these maps are safe to use as long as we aren't
-// writing to them from multiple threads at the same time. Since we only update these
-// from the single-threaded eventloop in userspace, we are ok.
-// See https://github.com/firezone/firezone/issues/10138#issuecomment-3186074350
-
-#[map]
-static CHAN_TO_UDP_44: HashMap<ClientAndChannelV4, PortAndPeerV4> =
-    HashMap::with_max_entries(NUM_ENTRIES, 0);
-#[map]
-static UDP_TO_CHAN_44: HashMap<PortAndPeerV4, ClientAndChannelV4> =
-    HashMap::with_max_entries(NUM_ENTRIES, 0);
-#[map]
-static CHAN_TO_UDP_66: HashMap<ClientAndChannelV6, PortAndPeerV6> =
-    HashMap::with_max_entries(NUM_ENTRIES, 0);
-#[map]
-static UDP_TO_CHAN_66: HashMap<PortAndPeerV6, ClientAndChannelV6> =
-    HashMap::with_max_entries(NUM_ENTRIES, 0);
-#[map]
-static CHAN_TO_UDP_46: HashMap<ClientAndChannelV4, PortAndPeerV6> =
-    HashMap::with_max_entries(NUM_ENTRIES, 0);
-#[map]
-static UDP_TO_CHAN_46: HashMap<PortAndPeerV4, ClientAndChannelV6> =
-    HashMap::with_max_entries(NUM_ENTRIES, 0);
-#[map]
-static CHAN_TO_UDP_64: HashMap<ClientAndChannelV6, PortAndPeerV4> =
-    HashMap::with_max_entries(NUM_ENTRIES, 0);
-#[map]
-static UDP_TO_CHAN_64: HashMap<PortAndPeerV6, ClientAndChannelV4> =
-    HashMap::with_max_entries(NUM_ENTRIES, 0);
-
-// Per-CPU data structures to learn relay interface addresses
-#[map]
-static INT_ADDR_V4: PerCpuArray<InterfaceAddressV4> = PerCpuArray::with_max_entries(1, 0);
-#[map]
-static INT_ADDR_V6: PerCpuArray<InterfaceAddressV6> = PerCpuArray::with_max_entries(1, 0);
-
-#[xdp]
-pub fn handle_turn(ctx: XdpContext) -> u32 {
-    try_handle_turn(&ctx).unwrap_or_else(|e| match e {
-        Error::NotIp | Error::NotUdp => xdp_action::XDP_PASS,
-
-        Error::InterfaceIpv4AddressAccessFailed
-        | Error::InterfaceIpv6AddressAccessFailed
-        | Error::PacketTooShort
-        | Error::NotTurn
-        | Error::NotAChannelDataMessage
-        | Error::Ipv4PacketWithOptions
-        | Error::UnsupportedChannel(_)
-        | Error::NoEntry(_) => {
-            debug!(&ctx, "Passing packet to userspace: {}", e);
-
-            xdp_action::XDP_PASS
-        }
-
-        Error::BadChannelDataLength | Error::XdpAdjustTailFailed(_) => {
-            warn!(&ctx, "Dropping packet: {}", e);
-
-            xdp_action::XDP_DROP
-        }
-    })
+// For non-Linux platforms: provide a stub main that exits with an error
```

Comment:
We don't want it on Linux though, the eBPF entrypoint is only valid on the bpf target :)

Reply:
ha, true! simplified it :)